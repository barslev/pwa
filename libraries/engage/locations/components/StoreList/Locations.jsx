import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import Store from './Store';
import { stores } from './style';

/**
 * Renders the locations where the product can be picked up.
 * @returns {JSX}
 */
function Locations({ context: Context }) {
  const { locations } = useContext(Context);

  if (!locations || locations.length === 0) {
    return null;
  }

  return (
    <div className={stores}>
      {locations.map(location => (
        <Store store={location} key={location.code} />
      ))}
    </div>
  );
}

Locations.propTypes = {
  context: PropTypes.elementType.isRequired,
};

export default Locations;
