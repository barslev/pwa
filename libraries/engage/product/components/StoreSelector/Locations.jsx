import React, { useContext } from 'react';
import { stores } from './style';
import Store from './Store';
import StoreSelectorContext from './context';

/**
 * Renders the locations where the product can be picked up.
 * @returns {JSX}
 */
function Locations() {
  const { locations } = useContext(StoreSelectorContext);

  if (!locations || locations.length === 0) {
    return null;
  }

  return (
    <div className={stores}>
      {locations.map(location => (
        <Store store={location} key={location.code} />
      ))}
    </div>
  );
}

export default Locations;
