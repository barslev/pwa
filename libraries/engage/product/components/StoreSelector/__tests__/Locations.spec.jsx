import React from 'react';
import { mount } from 'enzyme';
import Locations from '../Locations';
import StoreSelectorContext from '../context';

const locations = [
  {
    code: 1,
    name: 'test1',
  },
  {
    code: 2,
    name: 'test2',
  },
];

describe('<Locations />', () => {
  it('should not render if no locations have been passed', () => {
    const wrapper = mount((
      <StoreSelectorContext.Provider value={{}}>
        <Locations />
      </StoreSelectorContext.Provider>
    ));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance()).toEqual(null);
  });

  it('should render locations', () => {
    const wrapper = mount((
      <StoreSelectorContext.Provider value={{ locations }}>
        <Locations />
      </StoreSelectorContext.Provider>
    ));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Locations').getElements().length).toEqual(1);
  });
});
