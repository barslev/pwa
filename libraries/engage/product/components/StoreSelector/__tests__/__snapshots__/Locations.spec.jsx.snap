// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Locations /> should not render if no locations have been passed 1`] = `<Locations />`;

exports[`<Locations /> should render locations 1`] = `
<Locations>
  <div
    className={
      Object {
        "data-css-1ykq7o9": "",
      }
    }
  >
    <Store
      key="1"
      store={
        Object {
          "code": 1,
          "name": "test1",
        }
      }
    />
    <Store
      key="2"
      store={
        Object {
          "code": 2,
          "name": "test2",
        }
      }
    />
  </div>
</Locations>
`;
